PSCompatibilityCollector/Microsoft.PowerShell.CrossCompatibility/Collection/PowerShellDataCollector.cs

```diff
@@ -323,15 +344,31 @@ public PowerShellDataCollector Build(SMA.PowerShell pwsh, PowerShellVersion psVe
             moduleData.Cmdlets = cmdletData;
             moduleData.Functions = functionData;
 
+            if (aliasesToRequest != null && aliasesToRequest.Count > 0)
+            {
+                IEnumerable<AliasInfo> resolvedAliases = _pwsh.AddCommand(GcmInfo)
+                    .AddParameter("Name", aliasesToRequest)
+                    .InvokeAndClear<AliasInfo>();
+
+                foreach (AliasInfo resolvedAlias in resolvedAliases)
+                {
+                    if (resolvedAlias?.Definition == null)
+                    {
+                        continue;
+                    }
+
+                    aliases[resolvedAlias.Name] = resolvedAlias.Definition;
```

Comment:
Do we not need to check that `aliases` contains this entry before assigning to it?

Reply:
Well like in other places where we override an existing entry, we don't have much choice. Multiple modules can define the same alias. The real solution would be for us to collect multiple of them, but we're not currently set up for that. That's something I'd like to fix in the medium term

Reply:
I was rather thinking if we need a safe-guard around assigning on an index that might not exist
```csharp
if (aliases.ContainsKey(resolvedAlias.Name)
{
    aliases[resolvedAlias.Name] = resolvedAlias.Definition;
}
```

Reply:
I'm confused. We're just adding to the dictionary here; if we only add elements if the key is already there, we'll never add anything and end up with an empty dictionary, no?

Reply:
Forget what I said, everything's OK. I had a deja-vu of the bug in PR #1408 where I used the indexer to GET a dictionary entry, which throws if the key does not exist. In a similar way, `Dictionary.Add` throws an exception when the key already exists, but doing some research and a test taught me that using the indexer to SET the value does never throw no matter if the key already exists or not :-)